include/roboteam_ai/skills/NewPass.h

```diff
@@ -0,0 +1,58 @@
+//
+// Created by robzelluf on 1/22/19.
+//
+
+#ifndef ROBOTEAM_AI_PASS_H
+#define ROBOTEAM_AI_PASS_H
+
+#include <control/numTrees/NumTreePosControl.h>
+#include <control/BasicPosControl.h>
+#include "Skill.h"
+#include <coach/PassCoach.h>
+#include <control/PositionUtils.h>
+#include <utilities/Constants.h>
+#include <control/shotControllers/ShotController.h>
+
+namespace rtt {
+namespace ai {
+
+class NewPass : public Skill {
+protected:
+    enum PassType {
+        DEFAULT,
+        DEFENSIVE,
+        FREEKICK
+    };
+
+    PassType passType = DEFAULT;
+
+    PassType stringToType(const std::string& type);
+
+    const double CLOSE_ENOUGH_TO_BALL = 0.7;
+    const double SUCCESSFUL_PASS_ANGLE = 0.6;
+
+    bool forcePass = false;
```

Comment:
I see this is largely based on the previous pass skill. A Pass asks for a 'pass object' that will make your life much more simple.